src/libinjector/linux/methods/linux_write_file.c

```diff
@@ -298,13 +301,22 @@ event_response_t handle_write_file(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
 
 }
 
-static size_t read_chunk(injector_t injector, FILE* fp)
+static bool read_chunk(injector_t injector, size_t* bytes_read)
 {
-    injector->buffer.len = fread(injector->buffer.data, 1, FILE_BUF_SIZE, fp);
+    if (!bytes_read)
+        return false;
+
+    injector->buffer.len = fread(injector->buffer.data, 1, FILE_BUF_SIZE, injector->fp);
+    if (ferror(injector->fp))
+    {
+        PRINT_DEBUG("Failed to read from file\n");
```

Comment:
just print this error in `stderr`. I think it'll be helpful without debug build also